### DISCUSSION TITLE
Security question: What protections against regex injection attacks does json-schema-validator have?

### DISCUSSION THREAD

**Discussion by kbsizer** [json-schema-org/community#798] [Q&A] [answered]
Started to use json-schema-validator to meet a customer requirement.  A little bit of a learning curve, but -- so far -- the results are delightful.  Really impressed.   The only hiccup is a security concern about overly-complex and/or malicious regular expressions.   Are there any built-in constraints or safe-guards on regex size, complexity, or processing time?

Example of the type of thing I want to avoid or flag instead of hanging/crashing:
`{ "type" : "string", "pattern" : "^([\\w\\-\\.\\&\\,]+[\\s]*)+" }`

In general:  How to avoid or limit stuff like this: https://www.regular-expressions.info/catastrophic.html

Any help greatly appreciated!

**Comment by gregsdennis** [comment, marked as answer]
The validation spec, [section 10](https://json-schema.org/draft/2020-12/json-schema-validation#name-security-considerations), lists several security concerns, including regex.

Basically, it lists the concern and requires that implementations handle it themselves.  The idea is that the specification's role is to make tooling maintainers aware of the danger and allow them to manage it as required by best practices in their language/ecosystem.

**Reply by jdesrosiers** [reply]
@kbsizer In case it wasn't clear enough from Greg's answer... This repo is for the JSON Schema specification, not any particular implementation. You'll want to find the repo for the specific library you're using and ask them how/if they handle this case.

**Reply by kbsizer** [reply]
Thank you both!   And apologies:  If I'd thought half a second, that **_obviously_** had to be the case.  Duh.

**Reply by gregsdennis** [reply]
No worries.  It's not something that's obvious to a lot of people.
